### PR TITLE
Well known operators

### PR DESCRIPTION
**Pull request by stil4m**
* Adds code generation script for generating interfaces for packages that contain operators.
* Add option to include well known operators to the postprocesing by using `Elm.Processing.init |> Elm.Processing.withWellKnownOperators`

### CODE REVIEW

**Comment by jfmengels** [comment]
> @jfmengels Would this allow you to reduce your code base on elm-review a bit? Did I miss anything?

Yes it would. I do feel like this is a bit overkill, since the list is unlikely to change in an unpredictable manner, and because there is a lot of useless data in there that could be trimmed out.

FYI, here is pretty much the code that I have in `elm-review` to have the list of operators: https://ellie-app.com/8GCKWvGK8XLa1

**Comment by jfmengels** [comment]
I would personally imagine this concept of Processing and post-processing to be removed and that we'll end up with a `parse : String -> Result ...` function, as operators seem very stable. This would be more easier for the user, and would allow to only do a single pass for a file.

I get the feeling you wish to be more cautious or are more pessimistic about removing this concept.

File: src/Elm/Interface.elm [hunk 1]
```diff
@@ -55,6 +55,16 @@ type Exposed
     | Operator Infix
 
 
+{-| Type representing all infromation regarding an infix operator.
+-}
+type alias Infix =
```
[the remark this answers is not in the document]
Comment (stil4m) [reply src/Elm/Interface.elm:60]: From an interface perspective (i.e. functions from a dependency) you do not have Ranges on the elements defined in the `Infix`. From a source perspective, you do.

**Comment by stil4m** [comment]
Currently I leave processing in. Besides rebalancing operators and fixing the relation between documentation and declarations, it allows for smarter doing stuff like making lookup tables. For this you need a context with dependencies and other syntax files for which `parse : String -> Result ...` will just not suffice.

**Pull request merged**